sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -25,17 +25,26 @@ import org.apache.spark.sql.catalyst.util.TypeUtils
 import org.apache.spark.sql.types._
 import org.apache.spark.util.Utils
 
+trait PredResult
```

Comment:
predicate is only returns `true/false`, but also possible be `null`, however, `null` will be casted into `false` in scala, but this is important for `antijoin`.
```scala
scala> val a: Any = null
a: Any = null

scala> a.asInstanceOf[Boolean]
res0: Boolean = false
```